src/core/engine/processors/mongo_to_postgresql_order_status_history.py

```diff
@@ -0,0 +1,55 @@
+from src.constants.etl_status import EtlStatus
+from src.core.engine.processors.abstract_etl_processor import AbstractEtl
+from src.lib.entities.mongo_engine_odm_mapping import (
+    OrderStatusHistory as MongoOrderStatusHistory,
+)
+from src.lib.entities.sqlalchemy_orm_mapping import (
+    OrderStatusHistory as SqlalchemyOrderStatusHistory,
+)
+from src.utils.etl_util import (
+    convert_odm_order_status_history_to_orm_order_status_history,
+)
+
+
+class MongoToPostgresqlOrderStatusHistory(AbstractEtl):
+    def __init__(self, app_processor_config, app_context=None):
+        super().__init__(
+            app_processor_config=app_processor_config, app_context=app_context
+        )
+
+        self.mongo_connection = None
+        self.postgresql_session = None
+
+    def set_app_context(self, app_context):
+        ioc = app_context.ioc
+        self.mongo_connection = ioc.get_instance("mongo_engine_connection")
+        self.postgresql_session = ioc.get_instance("sqlalchemy_session")
+
+    def extract_data(self):
+
+        order_status_histories_from_mongo = MongoOrderStatusHistory.objects(
+            etl_status=EtlStatus.UNPROCESSED.value
+        )
+
+        return order_status_histories_from_mongo
+
+    def transform_data(self, extracted_data):
+        transformed_data = []
+
+        for order_status_history in extracted_data:
+            postgresql_order_status_history = SqlalchemyOrderStatusHistory()
+            postgresql_order_status_history_with_mongo_data = (
+                convert_odm_order_status_history_to_orm_order_status_history(
+                    order_status_history, postgresql_order_status_history
+                )
+            )
+            order_status_history.etl_status = EtlStatus.PROCESSED.value
+            order_status_history.save()
```

Comment:
This really bad! This should be happening in the repository!

Reply:
This should not happen when transforming the data!!! Should be happening when you load the data in the target